cli/src/pcluster/templates/slurm_builder.py

```diff
@@ -677,6 +675,30 @@ def _add_compute_resource_launch_template(
             ),
         )
 
+    def _generate_head_node_pass_role_resources(self):
+        """Return list of ARNs that head node ."""
+        default_pass_role_resource = self._format_arn(
+            service="iam",
+            region="",
+            resource=f"role{self._cluster_scoped_iam_path()}*",
+        )
+
+        # If there are any queues where a custom instance role was specified,
+        # enable the head node to pass permissions to those roles.
+        custom_queue_role_arns = [
+            arn for queue in self.config.scheduling.queues for arn in queue.iam.instance_role_arns
```

Comment:
What if the same custom role is specified in multiple queues? Is it ok to have redundant resources in the policy?

Reply:
I tested with a config containing duplicated instance role values in two different compute queues. Everything worked as expected, but I agree that this isn't the best way to handle this. The method implementing the `iam.instance_role_arns` param could remove duplicates first, but I think it also makes more sense to have `_generate_head_node_pass_role_resources` remove the duplicates, because there's a default value that may need to be added there.

That consideration also points out that the logic to figure out whether or not to include the default value currently uses the length of the value returned by `iam.instance_role_arns`. Since a given queue can have multiple instance role arns pointed to by a custom instance profile, that doesn't work. I'll change that as well so that it looks for any queues that don't specify any instance role or instance profile.

Reply:
Thank you for testing it out. Just out of curiosity, did IAM carry those duplicated resources or merge them together?

Reply:
The duplicates are not present in the template generated for the cluster stack. They are being removed sometime between when the resource policy construct is initialized and when the cdk app is synthesized. My guess is that happens when the construct is initialized but I don't know for sure.